### DISCUSSION TITLE
How to customise float window?

### DISCUSSION THREAD

**Discussion by augustocdias** [stevearc/overseer.nvim#9] [Q&A] [answered]
How do I change the highlights from the float windows?

I would like to change the background. I'd like to change the color and remove the transparency.

**Comment by stevearc** [comment, marked as answer]
There are three categories of floating windows that have config options. There's the "form" floats that open when you're launching or editing a task, the "confirm" float which only shows up when saving a task, and the "task_win" task output float that is opened from the "open float" action.

All of these are configured in the call to `setup`, and I believe the options you are looking for is to set `winblend = 0` (see `:help winblend`) and `winhighlight = "NormalFloat:MyCustomHighlight"` (see `:help winhighlight`).
```lua
require("overseer").setup({
  form = {
    win_opts = {
      winblend = 0,
      winhighlight = "NormalFloat:MyCustomHighlight",
    },
  },
  confirm = {
    win_opts = {
      winblend = 0,
      winhighlight = "NormalFloat:MyCustomHighlight",
    },
  },
  task_win = {
    win_opts = {
      winblend = 0,
      winhighlight = "NormalFloat:MyCustomHighlight",
    },
  },
})
```

**Reply by augustocdias** [reply]
Another question. Is there a way to focus on the float when I press `p` to open the preview of a task? Weirdly it is not focusing on that window and it is putting nvim in `insert` mode inside the task list window.

**Reply by augustocdias** [reply]
Isn't the preview window supposed to accept custom options too? 
https://github.com/stevearc/overseer.nvim/blob/2fb1794345ed174579014df5fe361dcc933c42f0/lua/overseer/task_list/sidebar.lua#L97-L107
It seems winblend, border, etc are all hardcoded.

**Reply by stevearc** [reply]
The "p" to preview is intended just for previewing the output. If you want your cursor to focus the window, use `<C-f>` to "open float" instead.

You're likely entering insert mode because you have an autocmd set up to do `startinsert` on `BufEnter` or some other event (it's recommended in the vim help docs, so a lot of people have that autocmd). If you can find out what it is, I might be able to give you some pointers on things you could adjust.

**Reply by stevearc** [reply]
The preview window opts were an oversight. Should be fixed now

**Reply by augustocdias** [reply]
> You're likely entering insert mode because you have an autocmd set up to do `startinsert` on `BufEnter` or some other event (it's recommended in the vim help docs, so a lot of people have that autocmd). If you can find out what it is, I might be able to give you some pointers on things you could adjust.

You're completely right. I have a start insert for when the buffer is a terminal. 

Do you set a buftype for the float or is there anything I can check to avoid this behavior for overseer popups?

**Reply by stevearc** [reply]
I _think_ the issue is that the autocmd event is running on BufEnter, BufWinEnter, or similar and it's triggering `startinsert` even though you're not in the terminal buffer.

I pushed up a change that opens the preview window with `noautocmd = true`, which might fix your issue. If not, I recommend some version of the config below:

```lua
vim.api.nvim_create_autocmd("TermOpen", {
  desc = "Auto enter insert mode when opening a terminal",
  pattern = "*",
  group = aug,
  callback = function()
    -- You could even wrap this in vim.defer_fn() if there's a possibility that
    -- the terminal loses focus immediately
    if vim.api.nvim_buf_get_option(0, "buftype") == "terminal" then
      vim.cmd([[startinsert]])
    end
  end,
})
```

**Reply by Kailian-Jacy** [reply]
@stevearc Is there any way we can Identify the floating window buffer, like buftype = "OverseerFloat" or something? 

On top of this auto-inserting condition, that may also help to customize buffer local keymaps to the floating buffer, like bind "q" to hide the buffer, and like unset the keymap for delete the buffer (which always makes me lose task log).
